src/System.Text.Formatting/System/Text/Parsing/SequenceParser.cs

```diff
@@ -37,47 +37,48 @@ public static class TextSequenceExtensions
-                    free = free.Slice(second.Length);
-
-                    ReadOnlyBuffer<byte> next;
-                    while (free.Length > 0) {
-                        if (bufferSequence.TryGet(ref position, out next)) {
-                            if (next.Length > free.Length) next = next.Slice(0, free.Length);
-                            next.CopyTo(free);
-                            free = free.Slice(next.Length);
-                        }
-                        else {
-                            break;
-                        }
+                Span<byte> destination = stackalloc byte[StackBufferSize];
+
+                first.CopyTo(destination);
+                var free = destination.Slice(first.Length);
+
+                if (second.Length > free.Length) second = second.Slice(0, free.Length);
+                second.CopyTo(free);
+                free = free.Slice(second.Length);
+
+                ReadOnlyBuffer<byte> next;
+                while (free.Length > 0)
+                {
+                    if (bufferSequence.TryGet(ref position, out next))
+                    {
+                        if (next.Length > free.Length) next = next.Slice(0, free.Length);
+                        next.CopyTo(free);
+                        free = free.Slice(next.Length);
                     }
+                    else
+                    {
+                        break;
+                    }
+                }
 
-                    combinedSpan = destination.Slice(0, StackBufferSize - free.Length);
+                var combinedStackSpan = destination.Slice(0, StackBufferSize - free.Length);
 
-                    // if the stack allocated buffer parsed succesfully (and for uint it should always do), then return success. 
-                    if (PrimitiveParser.InvariantUtf8.TryParseUInt64(combinedSpan, out value, out consumed)) {
-                        if(consumed < combinedSpan.Length || combinedSpan.Length < StackBufferSize) {
-                            return true;
-                        }
+                // if the stack allocated buffer parsed succesfully (and for uint it should always do), then return success. 
+                if (PrimitiveParser.InvariantUtf8.TryParseUInt64(combinedStackSpan, out value, out consumed))
+                {
+                    if (consumed < combinedStackSpan.Length || combinedStackSpan.Length < StackBufferSize)
+                    {
+                        return true;
                     }
                 }
             }
 
             // for invariant culture, we should never reach this point, as invariant uint text is never longer than 127 bytes. 
             // I left this code here, as we will need it for custom cultures and possibly when we shrink the stack allocated buffer.
-            combinedSpan = bufferSequence.ToSpan();
-            if (!PrimitiveParser.InvariantUtf8.TryParseUInt64(first.Span, out value, out consumed)) {
```

Comment:
Was 'first.Span' a bug here?

Reply:
I am not sure but it seems like it given combinedSpan is assigned and then not used.
cc @KrzysztofCwalina

Reply:
I would not be shocked if this codepath never executed (as the comment implies)